### PR TITLE
Guide to developing homebrew for repro carts

### PR DESCRIPTION
**Pull request by velipso**
There have been discussions/questions on discord around how bootleg carts work, and how to save on them.  This is intended to be a guide for homebrew developers.  More could definitely be added to it, but I think it's a good starting point.

### CODE REVIEW

**Review by avivace — CHANGES_REQUESTED** [review]
I really like this! Thanks @velipso . My only concern is the mentioning / direct pointers and screenshots of Aliexpress..

**Review by avivace — APPROVED** [review]
Looks great to me, thanks!

File: content/bootleg-carts/introduction.md [hunk 1]
```diff
@@ -0,0 +1,270 @@
...
+`0xA9` -- this is because D0/D1 are swapped (`10101010` -> `10101001`)!  [See the section
+above](#swapping-of-d0d1).
+
+So if we want to perform a reset on the chip, we just write `0xF0` to any address.  Note that reset
+doesn't _erase_ the chip, it just resets any commands in progress.
+
+```c
+// reset
+*((u16 *)0x08000000) = 0xF0;
+__asm("nop");
```
Comment (exelotl) [inline content/bootleg-carts/introduction.md:165]: Is it possible to use `WAITCNT` + ROM mirrors to eliminate the need for `nop`? If so, it might be worth mentioning and/or demonstrating what settings to use.

File: content/bootleg-carts/introduction.md [hunk 2]
```diff
@@ -0,0 +1,270 @@
...
+
+// reset
+_FLASH_WRITE(0, 0xF0);
+// auto-select
+_FLASH_WRITE(0xAAA, 0xA9);
+_FLASH_WRITE(0x555, 0x56);
+_FLASH_WRITE(0xAAA, 0x90);
+```
+
+**IMPORTANT NOTE:** Since we need to control the reads/writes sent to the ROM, we cannot run the
```
Comment (exelotl) [inline content/bootleg-carts/introduction.md:182]: I still don't understand the technical reason for this, is it just the nature of flash chips that they can't be read from while they are handling commands? Is there a way to reword this to make the _why_ more clear for dummies like me?

**Review by exelotl — APPROVED** [review]
It all looks good to me! I've left some comments, but can always be addressed in a future PR.

File: content/bootleg-carts/introduction.md (hunk 1 above)
Comment (velipso) [reply content/bootleg-carts/introduction.md:165]: That's a good question, I don't know.  I haven't tested that.

File: content/bootleg-carts/introduction.md (hunk 2 above)
Comment (velipso) [reply content/bootleg-carts/introduction.md:182]: Some operations on the flash chips (CFI and Auto-Select are two that come to mind) actually change the read behavior of the chip.  You don't want the CPU reading special values from the ROM while trying to execute.

File: content/bootleg-carts/introduction.md [hunk 3]
```diff
@@ -0,0 +1,270 @@
...
+
+### Querying for Information
+
+There is a standard protocol used by all flash chips called the [Common Flash Memory
+Interface](https://en.wikipedia.org/wiki/Common_Flash_Memory_Interface) (CFI).
+
+You can use CFI to query a lot of information about the chip you're interacting with.  The chip
+specifications should have a section on CFI.
+
+TODO: write more information about CFI and sector layout.
```
[the remark this answers is not in the document]
Comment (velipso) [reply content/bootleg-carts/introduction.md:195]: Yeah, I'm happy to add more when I get it figured out myself, or someone else can too.  Goombacolor doesn't solve this issue in a general way, so I couldn't link to it in this case.

**Comment by velipso** [comment]
I think we're good to merge this for now..?  I know it isn't perfect but I think it's a good start and I can add to it over time, if that's okay.

File: content/bootleg-carts/introduction.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (avivace) [reply content/bootleg-carts/introduction.md:195]: Maybe we can open an issue about this?
